packages/arb-token-bridge-ui/src/tokenLists.tsx

```diff
@@ -157,34 +156,37 @@ export function fetchTokenLists(): Promise<void> {
           }
         })
 
-      storage = JSON.stringify(tokenListsWithBridgeTokenListId)
-      resolve()
+      resolve(tokenListsWithBridgeTokenListId as TokenListWithId[])
     })
   })
 }
 
-export function useTokenLists(forL2ChainId?: string): TokenListWithId[] {
-  const [tokenLists, setTokenLists] = useState<TokenListWithId[]>(() =>
-    getTokenLists(forL2ChainId)
-  )
-
-  useEffect(() => {
-    setTokenLists(getTokenLists(forL2ChainId))
-  }, [forL2ChainId])
-
-  return tokenLists
-}
-
-export function getTokenLists(forL2ChainId?: string): TokenListWithId[] {
-  if (!storage) {
-    return []
-  }
-
-  const parsedStorage: TokenListWithId[] = JSON.parse(storage)
+export function useTokenLists(forL2ChainId?: string): {
+  tokenLists: TokenListWithId[]
+  isFetching: boolean
+} {
+  const { data = [], isValidating } = useSWRImmutable(
+    ['useTokenLists', forL2ChainId],
+    async () => {
+      let newTokensList = []
+      const result = await fetchTokenLists(forL2ChainId)
+
+      if (typeof forL2ChainId === 'undefined') {
+        newTokensList = result
+      } else {
+        newTokensList = result.filter(
+          tokenList => tokenList.l2ChainId === forL2ChainId
+        )
+      }
 
-  if (typeof forL2ChainId === 'undefined') {
-    return parsedStorage
-  }
+      return newTokensList
+    },
+    {
+      shouldRetryOnError: true,
+      errorRetryCount: 2,
+      errorRetryInterval: 1_000
+    }
+  )
 
-  return parsedStorage.filter(tokenList => tokenList.l2ChainId === forL2ChainId)
+  return { tokenLists: data, isFetching: isValidating }
```

Comment:
Why not just return everything from `useSWRImmutable`, meaning: 

```
export function useTokenLists(forL2ChainId?: string) {
  return useSWRImmutable(...)
}
```

That way we'll have the same shape of data in every hook that uses `useSWR`.